sympy/solvers/ode.py

```diff
@@ -2318,9 +2320,20 @@ def _expand(expr):
     # We cleaned up the constants before solving to help the solve engine with
     # a simpler expression, but the solved expression could have introduced
     # things like -C1, so rerun constantsimp() one last time before returning.
+
+    # Don't clobber the symbols already in the ODE. This is important when the
+    # input ODE derives from output of a previous call to dsolve (issue #4838)
+    if iterable(eq):
+        symbols_eq = set()
```

Comment:
this is not currently used (though I kind of like this idea of using `exclude`)

Reply:
I guess I missed "general cleanup" off the TODO list. I want to get agreement on the signatures first though.

Reply:
That's left over from the previous version of this PR.